File: ProjetMP3/src/client/ClientMP3.cpp
#include <Ice/Ice.h>
#include <InterfaceMP3.h>

using namespace std;
using namespace ArchDistrib;

int main(int argc, char* argv[]) {
    int status = 0;
    Ice::CommunicatorPtr ic;

    try {
        ic = Ice::initialize(argc, argv);
        Ice::ObjectPrx base = ic->stringToProxy("InterfaceMP3:default -p 10000");
        InterfaceMP3Prx serveur = InterfaceMP3Prx::checkedCast(base);
        if (!serveur)
            throw "Invalid proxy";

        std::string mes;
        cin >> mes;
        while (mes != "exit") {
            serveur->printString(mes);
            cin >> mes;
        }
    }
    catch (const Ice::Exception& ex) {
        cerr << ex << endl;
        status = 1;
    }
    catch (const char* msg) {
        cerr << msg << endl;
        status = 1;
    }

    if (ic)
        ic->destroy();

    return status;
}

File: ProjetMP3/src/client/ClientMP3.py
import sys, traceback, Ice
import ArchDistrib
import CLI
#import GUI


SERVER_IP = "localhost"

def main():
    status = 0
    ic = None
    try:
        # Configuration pour augmenter la taille des requetes,
        # afin de pouvoir envoyer une chanson entiere
        #     20480 KB permet d'envoyer une grande chanson de 20 MB
        props = Ice.createProperties(sys.argv)
        props.setProperty("Ice.MessageSizeMax", "20480")
        iniData = Ice.InitializationData()
        iniData.properties = props

        # Initialisation du serveur et Ice
        ic = Ice.initialize(iniData)
        base = ic.stringToProxy("InterfaceMP3:default -h " + SERVER_IP + " -p 10000")
        serveur = ArchDistrib.InterfaceMP3Prx.checkedCast(base)
        if not serveur:
            raise RuntimeError("Invalid proxy")

        # Client
        client = CLI.App(serveur)
        #client = GUI.App(serveur)
        #client = GUI.App(None)

        # Boucle principale
        client.mainloop()

    except:
        traceback.print_exc()
        status = 1

    if ic:
        # Clean up
        try:
            ic.destroy()
        except:
            traceback.print_exc()
            status = 1

    sys.exit(status)


if __name__ == "__main__":
  main()
